src/interceptors/add_multitenant_client_secret_to_params.js

```diff
@@ -0,0 +1,31 @@
+/**
+   Read `multitenantClientSecret` and `hostname` from `ctx` and add signed JWT
+   as `clientSecret` to `params`.
+
+   Changes to `ctx`:
+
+   - add `params.client_secret`
+*/
+import jwt from 'jsonwebtoken';
+
+const signingOpts = {
+  algorithm: 'HS256',
+  expiresIn: 60, // seconds, should be enough for possible clock skew, but not too long
+};
+
+export default class AddMultitenantClientSecretToParams {
```

Comment:
Maybe it would be better to split this into two interceptors, because for the `GET /client_data` endpoint, the secret needs to be passed as a header.